### ISSUE TITLE
Context repetition unlikelihood loss for a ranker model

### ISSUE THREAD

**Issue by samuelazran** [facebookresearch/ParlAI#3525]
While fine-tuning poly-encoder (ranker) model on dialogue datasets I see it suffers from the issue of context repetition just like generative models (https://parl.ai/projects/dialogue_unlikelihood/). The issue occurs when the context contain tokens from the candidate response, in those cases, the model ranks those responses higher than the correct response seen in the training dataset.

Is it possible to use or adjust the above context repetition unlikelihood loss for a ranker model training? if so could anyone guide me in how to go about it? I think it can be helpful for many use-cases. 

Basically, I would like to have a bigger loss in case the model miss-rank as the top response a candidate that contain certain amount of n-grams from the context.

I'm thankful for any help.

**Comment by stephenroller** [comment]
Ranking models also have a hard repetition blocking limit

https://github.com/facebookresearch/ParlAI/blob/00286c5f3e9ef7ea63797b692430cad9fe79babf/parlai/core/torch_ranker_agent.py#L89-L96

As far as UL for ranking models, that would be a whole research question, and I don't think we can support you directly on that

**Comment by samuelazran** [comment]
@stephenroller thank you so much for the quick and elaborate response. 

I know about the repetition blocking limit for ranking models. However it doesn't suit my use-case where in some cases there is repetition in the training data. 

If it is possible to incorporate in the loss function of a ranking model the overlap between the context and the top response candidate tokens I'd love to see some examples and share my experiments with it.
